### ISSUE TITLE
Migration from react-firebase-hooks

### ISSUE THREAD

**Issue by mschmid** [skillnation/react-native-firebase-hooks#2]
Great to see a library like react-firebase-hooks supporting react-native!

When just replacing `react-firebase-hooks` with `react-native-firebase-hooks` on an existing app (already in production), I just get infinite loading screens for:

```
  const [profile, profileLoading, profileError] = useDocumentData(
    userAuth && firestore().collection('users').doc(userAuth.uid),
    {
      snapshotListenOptions: { includeMetadataChanges: true },
    },
  )
  ```
  
`profileLoading` just remains `true`.

Is there any cache that needs to be deleted for migration, or any change in the API I missed?
Thanks for any help. Would love to use react-native-firebase-hooks for the possibility to use other libraries that depend on Firebase V9.

**Comment by hannojg** [comment]
Hey!

Which version are you on?

The issue is very likely that an "un-memoized" object is being passed to the hook. Change it for example to smth like this:

```js
const options = {
    snapshotListenOptions: { includeMetadataChanges: true },
  };

function Component() {
  const [profile, profileLoading, profileError] = useDocumentData(
    userAuth && firestore().collection('users').doc(userAuth.uid),
    options,
  )
}
```

However, this is indeed a bug and you should be able to just pass the object "inline" without having to memorize it.
Will fix this.

Please let me know if this workaround works or no, thank you ✌️

**Comment by mschmid** [comment]
I am using `"@skillnation/react-native-firebase-hooks": "0.4.0"`.

Unfortunately, "memoizing" options or even removing options did not do the trick.

Interestingly, it works now for the first query (`useDocumentData`, fetching the profile), but not for the second (`useCollectionData` pulling users documents).

In the iOS Simulator I also get it running by saving the source file, which triggers a refreshing of the app. Re-rendering it by changing state of an unrelated variable however doesn't do the trick.

Reloading the whole app with the React Native Debug Menu then leads to the infinite loading screen again.

**Comment by hannojg** [comment]
Okay, I have to reproduce this to find a fix for that. One solution might be to use `useCollectionDataOnce` in the meantime.
I'll see if I find some time on the weekend to fix it - it would speed up things if you could create a reproduction, or even dig into the code and create a PR ✌️

**Comment by redpandatronicsuk** [comment]
I have a similar bug.

```js
const [
    data,
    dataLoading,
    dataError,
  ] = useCollectionData(
    firestore()
      .collection(`…`)
      .where('createdAt', '>', startOfMonth)
      .where('createdAt', '<=', endOfMonth)
  );
```

loads data the first time, but when `startOfMonth` or `endOfMonth` change `dataLoading` will always be true. I.e. adding `console.log(dataLoading, data, dataError)` would print:

```js
undefined true undefined
[…] false undefined
// Now changing startOfMonth and endOfMonth to load new data:
undefined true undefined
// and no more data is loaded now, even when reverting startOfMonth and endOfMonth back to their initial values will only give:
undefined true undefined
```

**Comment by hannojg** [comment]
Hey,

sorry, i totally forgot about that issue. Will find some time this week to look into it ✌️

**Comment by hannojg** [comment]
Hey, I was able to reproduce the issue in the example app (had to modernize that one first).
Will provide a fix shortly ✌️

**Comment by hannojg** [comment]
[0.4.1](https://github.com/skillnation/react-native-firebase-hooks/releases/tag/v0.4.1) has just been released, and it should be fixed. Can you please try? Will reopen if there are still issues

// Edit: The issue was also present for `use*Once` hooks, this is fixed now as well in [0.5.0](https://github.com/skillnation/react-native-firebase-hooks/releases/tag/v0.5.0)

**Comment by redpandatronicsuk** [comment]
Thanks, seems to work now!
